pkg/target/manager.go

```diff
@@ -317,22 +317,21 @@ func (m *managerImpl) TargetControlPlane(ctx context.Context) error {
 	return m.updateTarget(target)
 }
 
-func (m *managerImpl) UnsetTargetControlPlane() (string, error) {
+func (m *managerImpl) UnsetTargetControlPlane() error {
 	currentTarget, err := m.CurrentTarget()
 	if err != nil {
-		return "", fmt.Errorf("failed to get current target: %v", err)
+		return fmt.Errorf("failed to get current target: %v", err)
 	}
 
-	targetedName := currentTarget.ShootName()
-	if currentTarget.ControlPlaneFlag() {
-		return targetedName, m.patchTarget(func(t *targetImpl) error {
-			t.ControlPlane = false
+	if currentTarget.ControlPlane() {
+		return m.patchTarget(func(t *targetImpl) error {
+			t.ControlPlaneFlag = false
 
 			return nil
 		})
 	}
 
-	return "", ErrNoControlPlaneTargeted
+	return ErrNoControlPlaneTargeted
```

Comment:
```go
	if !currentTarget.ControlPlane() {
		return ErrNoControlPlaneTargeted
	}

	return m.patchTarget(func(t *targetImpl) error {
		t.ControlPlaneFlag = false

		return nil
	})
```

I would turn it around, but this is just my personal preference. Feel free to ignore

Reply:
ok